faraday/proxy.py

```diff
@@ -393,6 +428,208 @@ def testdb_read_worker():
+        dataQueue = []
+        try:
+            dataQueue = getDicts[unit][1]
+        except KeyError as e:
+            # Simply haven't ever received data so break
+            pass
+
+        if len(dataQueue) <= 0:
+            conn.sendall("No Data! Goodbye.")
+            closeConnection(conn, addr, unit)
+            break
+
+        try:
+            # pop off a data entry from the left of getDicts
+            temp = dataQueue.popleft()
+
+        except IndexError as e:
+            # Empty queue
+            logger.error(e)
+
+        try:
+            data = temp['data'].decode('base64', 'strict')
+
+        except StandardError as e:
+            logger.error(e)
+
+        # unpack frames and retrieve data originally sent to socket
+        try:
+            dataList = struct.unpack("BB121s", data)
+            dataList2 = struct.unpack("B{0}s".format(payloadSize), dataList[2][:payloadSize + 1])
+            socketData = dataList2[1][:dataList2[0]]
+
+        except struct.error as e:
+            logger.error(e)
+
+        else:
+            try:
+                conn.sendall(socketData)
+
+            except socket.error as e:
+                closeConnection(conn, addr, unit)
+                break
+
+
+def socket_worker(modem, getDicts, dataPort, dataBuffer):
+    """
+    Create sockets for data connections
+
+    """
+    logger.debug('Starting socket_worker thread')
+    unit = modem['unit']
+    dataBuffer[unit] = {}
+
+    # Start a socket server for the modem
+    server = startServer(unit, dataPort)
+
+    # Listen to server in infinit loop
+    server.listen(5)
+    while True:
+        # continuously accept connections and read data from socket to buffers
+        conn, addr = acceptConnection(server, unit)
+        receiveData(conn, addr, dataBuffer, unit)
+
+
+def socket_worker_RX(modem, getDicts, dataPort, dataBuffer, payloadSize):
+    """
+    Create sockets for data connections
+
+    """
+    logger.debug('Starting socket_worker thread')
+    unit = modem['unit']
+    dataBuffer[unit] = {}
+
+    # Start a socket server for the modem
+    server = startServer(unit, dataPort)
+
+    # Listen to server in infinit loop
+    server.listen(5)
+    while True:
+        # continuously accept connections and send data to socket from get buffer
+        conn, addr = acceptConnection(server, unit)
+        sendData(conn, addr, getDicts, unit, payloadSize)
+
+
+def createPacket(data, size):
+    # initialize temp variable list and packet
+    temp = []
+    packet = ''
+
+    # Pop off "size" bytes and append to temporary list
+    i = 0
+    for i in range(size):
+        try:
+            a = data.popleft()
+            temp.append(a)
+
+        except IndexError:
+            # simply an empty queue
+            pass
+    # Join list together and append two control bytes, convert to BASE64
+    try:
+
+        payload = ''.join(temp)
+        preamble = struct.pack("BB", 0, 0)  # Preamble is two 0's at this time
```

Comment:
Like above, these packet struct formats should probably be located at the global level as organized single locations to see both TX and RX formatters.